File: PRACTICA#2/HERENCIA/1/PYTHON/Moto.py
from Vehiculo import Vehiculo

class Moto(Vehiculo):
    def __init__(self, marca, modelo, año, precio_base, cilindrada, tipo_motor):
        super().__init__(marca, modelo, año, precio_base)
        self.__cilindrada = cilindrada
        self.__tipo_motor = tipo_motor
#A
    def get_cilindrada(self):
        return self.__cilindrada
    def set_cilindrada(self, cilindrada):
        self.__cilindrada = cilindrada
    def get_tipo_motor(self):
        return self.__tipo_motor
    def set_tipo_motor(self, tipo_motor):
        self.__tipo_motor = tipo_motor
    
    def mostrar_info(self):
        super().mostrar_info()
        print(f"Cilindrada: {self.__cilindrada}, Tipo de motor: {self.__tipo_motor}")
        

File: PRACTICA#2/HERENCIA/1/JAVA/Moto.java
public class Moto extends Vehiculo {
    private int cilindrada;
    private String tipoMotor;

    //A
    public Moto(String marca, String modelo, int año, double precioBase, int cilindrada, String tipoMotor) {
        super(marca, modelo, año, precioBase);
        this.cilindrada = cilindrada;
        this.tipoMotor = tipoMotor;
    }

    public int getCilindrada() {
        return cilindrada;
    }

    public void setCilindrada(int cilindrada) {
        this.cilindrada = cilindrada;
    }

    public String getTipoMotor() {
        return tipoMotor;
    }

    public void setTipoMotor(String tipoMotor) {
        this.tipoMotor = tipoMotor;
    }

    @Override
    public void mostrarInfo() {
        System.out.println("Marca: " + getMarca() + ", Modelo: " + getModelo() + ", Año: " + getAño() + ", Precio Base: " + getPrecioBase());
        System.out.println("Cilindrada: " + cilindrada + ", Tipo de Motor: " + tipoMotor);
    }
}